### ISSUE TITLE
get_fact - error on cisco virl xrv

### ISSUE THREAD

**Issue by amb1s1** [napalm-automation/napalm-iosxr#104]
### Description of Issue/Question
When i try to get_facts I get an error. I'm able to connect and I tested getting bgp neighbor without any issues.
### Did you follow the steps from https://github.com/napalm-automation/napalm#faq
[X ] Yes
[ ] No

### napalm-iosxr version
(Paste verbatim output from `pip freeze | grep napalm-iosxr` between quotes below)

```
napalm-iosxr (0.4.7)
```

### IOS-XR version and platform details
(Paste the complete verbatim output from `show version brief` between quotes below)

```
Sat Feb  4 20:57:25.341 UTC

Cisco IOS XR Software, Version 6.0.1[Default]
Copyright (c) 2016 by Cisco Systems, Inc.

ROM: GRUB, Version 1.99(0), DEV RELEASE

iosxrv-1 uptime is 1 hour, 10 minutes
System image file is "bootflash:disk0/xrvr-os-mbi-6.0.1/mbixrvr-rp.vm"

cisco IOS XRv Series (Pentium Celeron Stepping 3) processor with 3169791K bytes of memory.
Pentium Celeron Stepping 3 processor at 3113MHz, Revision 2.174
IOS XRv Chassis

1 Management Ethernet
2 GigabitEthernet
97070k bytes of non-volatile configuration memory.
866M bytes of hard disk.
2321392k bytes of disk0: (Sector size 512 bytes).
```

### Error Traceback
(Paste the complete traceback of the exception between quotes below)

from napalm import get_network_driver
driver = get_network_driver('iosxr')
device = driver('172.16.1.101', 'cisco', 'cisco')
device.open()
facts=device.get_facts()

```
>>> device.get_facts()
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/Library/Python/2.7/site-packages/napalm_iosxr/iosxr.py", line 154, in get_facts
    platform_attr_tree = facts_rpc_reply.xpath(platform_attr_xpath)[0]
IndexError: list index out of range
```

**Comment by 2stacks** [comment]
**Description of Issue/Question** - unable to use Description of Issue/Question with iosxr

**Did you follow the steps** - yes

**napalm-iosxr version** - napalm-iosxr==0.4.8

**IOS-XR version and platform details**

Mon Mar 20 01:30:22.614 UTC

Cisco IOS XR Software, Version 6.0.1[Default]
Copyright (c) 2016 by Cisco Systems, Inc.

ROM: GRUB, Version 1.99(0), DEV RELEASE

ciosxr-01 uptime is 3 hours, 38 minutes
System image file is "bootflash:disk0/xrvr-os-mbi-6.0.1/mbixrvr-rp.vm"

cisco IOS XRv Series (Pentium Celeron Stepping 3) processor with 3145215K bytes of memory.
Pentium Celeron Stepping 3 processor at 2631MHz, Revision 2.174
IOS XRv Chassis

1 Management Ethernet
3 GigabitEthernet
97070k bytes of non-volatile configuration memory.
866M bytes of hard disk.
2321392k bytes of disk0: (Sector size 512 bytes).


**Steps to Reproduce the Issue**

Using napalm-ansible

  tasks:
  - name: get facts from device
    napalm_get_facts:
      hostname={{ host }}
      username={{ username }}
      password={{ password }}
      dev_os={{ dev_os }}
      filter='facts,interfaces,bgp_neighbors'
    register: result

  - name: print data
    debug: var=result


**Error Traceback**

TASK [get facts from device] ***************************************************

fatal: [ciosxr-01]: FAILED! => {"changed": false, "failed": true, "msg": "[facts] cannot retrieve device data: list index out of range"}

**Comment by mirceaulinic** [comment]
@amb1s1 @2stacks is any of you guys able to submit a fix for this issue? Thanks!

**Comment by mekisiel** [comment]
I also ran into this issue using get_facts.

From the XR device the following is output when walking the xml-schema:

`xml-schema[oper]:$able.Rack.Attributes.BasicInfo> pwd`
`PlatformInventory.RackTable.Rack.Attributes.BasicInfo`
`xml-schema[oper]:$able.Rack.Attributes.BasicInfo> get`
`%MGBL-SCHBRO-7-DATAARRAYLENGTH : [xml-schema] Too little or too much naming information provided for class Rack `

In iosxr.py, using a debug print command for platform_attr_tree the following is output:
`[]`

The empty list explains `IndexError: list index out of range`.

Removing index `[0]` from the end of `platform_attr_tree = facts_rpc_reply.xpath(platform_attr_xpath)` fixes what I was originally after, which was a list of interfaces.

I can submit a PR, if you'd like. I'm not sure if this breaks anything else - have not had the time to test thoroughly.

**Comment by mirceaulinic** [comment]
Hi @mekisiel. Yes, a PR to fix this would be very welcome, thank you!

**Comment by mirceaulinic** [comment]
This will be solved in 0.5.6 thanks to @mekisiel
